llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -3666,6 +3666,63 @@ LegalizerHelper::bitcastConcatVector(MachineInstr &MI, unsigned TypeIdx,
   return Legalized;
 }
 
+/// This attempts to bitcast G_EXTRACT_SUBVECTOR to CastTy.
+///
+///  <vscale x 8 x i1> = G_EXTRACT_SUBVECTOR <vscale x 16 x i1>, N
+///
+/// ===>
+///
+///  <vscale x 2 x i1> = G_BITCAST <vscale x 16 x i1>
+///  <vscale x 1 x i8> = G_EXTRACT_SUBVECTOR <vscale x 2 x i1>, N / 8
+///  <vscale x 8 x i1> = G_BITCAST <vscale x 1 x i8>
+LegalizerHelper::LegalizeResult
+LegalizerHelper::bitcastExtractSubvector(MachineInstr &MI, unsigned TypeIdx,
+                                         LLT CastTy) {
+  auto ES = dyn_cast<GExtractSubvector>(&MI);
```

Comment:
How could this ever fail? The caller checked the opcode. Do all bitcast handlers do this? Or just G_CONCAT_VECTORS?

Reply:
It was in bitcastConcatVector. I thought the same thing, but felt it was safer to add it.

Reply:
If its not possible to hit we should just use `cast`. If it is possible, then it should have a test. Otherwise it's an untested path.

Reply:
Since we don't have a test, I'm changing it to cast.